optuna/integration/tensorboard.py

```diff
@@ -56,34 +56,36 @@ def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -
     def _add_distributions(
         self, distributions: Dict[str, optuna.distributions.BaseDistribution]
     ) -> None:
+        real_distributions = (
+            optuna.distributions.UniformDistribution,
+            optuna.distributions.LogUniformDistribution,
+            optuna.distributions.DiscreteUniformDistribution,
+        )
+        int_distributions = (optuna.distributions.IntUniformDistribution,)
+        categorical_distributions = (optuna.distributions.CategoricalDistribution,)
+        supported_distributions = (
+            real_distributions + int_distributions + categorical_distributions
+        )
+
         for param_name, param_distribution in distributions.items():
-            if isinstance(param_distribution, optuna.distributions.UniformDistribution):
-                self._hp_params[param_name] = hp.HParam(
-                    param_name, hp.RealInterval(param_distribution.low, param_distribution.high)
-                )
-            elif isinstance(param_distribution, optuna.distributions.LogUniformDistribution):
-                self._hp_params[param_name] = hp.HParam(
-                    param_name, hp.RealInterval(param_distribution.low, param_distribution.high)
-                )
-            elif isinstance(param_distribution, optuna.distributions.DiscreteUniformDistribution):
+            if isinstance(param_distribution, real_distributions):
                 self._hp_params[param_name] = hp.HParam(
-                    param_name, hp.RealInterval(param_distribution.low, param_distribution.high)
+                    param_name,
+                    hp.RealInterval(float(param_distribution.low), float(param_distribution.high)),
```

Comment:
I changed my mind to perform a type conversion in `tensorbaord.py` instead of initializers in `UniformDistribution` and `LogUniformDistribution` since I found some tests were failed (e.g. https://github.com/optuna/optuna/blob/f7469095b42463aaa6db87f22f46423ba3c63b48/tests/integration_tests/test_mlflow.py#L73) and wondered if some users will be affected by this change.

@HideakiImamura How do you think about it?

Reply:
In my understanding, Optuna discourages users to use the distribution classes directly, so casting float inside the classes sounds fine.

Reply:
I think it's better to convert the type in `tensorboard.py` since it reduces the number of changed files.

Reply:
I fixed code to do (a). To follow up this PR, we can merge it
https://github.com/optuna/optuna/pull/2360